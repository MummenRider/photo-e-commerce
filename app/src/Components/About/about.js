import React from "react";
import {
  Container,
  Image,
  Content,
  Title,
  Description,
  Inner,
  Frames,
  Left,
  Right,
} from "Components/About/about.styles";
import { useInView } from "react-intersection-observer";

export default function About({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

About.Image = function AboutImage({ src, ...restProps }) {
  const [ref, inView] = useInView({
    rootMargin: "0px 0px -30% 0px",
  });
  return (
    <Frames {...restProps}>
      <Image
        url={src}
        ref={ref}
        animate={{
          scale: inView ? 1.0 : 1.1,
        }}
        transition={{
          ease: [0.43, 0.13, 0.23, 0.96],
          duration: 0.8,
        }}
      />
    </Frames>
  );
};

About.Content = function AboutContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

About.Left = function AboutLeft({ children, ...restProps }) {
  return <Left {...restProps}>{children}</Left>;
};

About.Right = function AboutRight({ children, ...restProps }) {
  return <Right {...restProps}>{children}</Right>;
};

About.Title = function AboutTitle({ isBigDevice, children, ...restProps }) {
  return isBigDevice ? <Title {...restProps}>{children}</Title> : null;
};

About.Description = function AboutDescription({
  options,
  children,
  ...restProps
}) {
  const [ref, inView] = useInView({ ...options });

  return (
    <Description
      ref={ref}
      {...restProps}
      animate={{
        y: inView ? 0 : 100,
        opacity: inView ? [0.2, 0.4, 0.8, 1] : 0,
      }}
      transition={{
        ease: [0.43, 0.13, 0.23, 0.96],
        duration: 1,
      }}
    >
      {children}
    </Description>
  );
};
