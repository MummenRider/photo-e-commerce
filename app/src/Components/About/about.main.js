import {
  AboutContainer,
  Description,
  Image,
  ImageOne,
  ImageTwo,
  LargeQuote,
  MidQuote,
  ImageFive,
  ImageFour,
  ImageThree,
  SmallQuote,
  HeroStack,
  ImageDivider,
  ImageFourContainer,
  ImageDescription,
  ImageFiveContainer,
  TextContent,
  AboutTitle,
} from "Components/About/about.main.styles.js";
import useMeasure from "react-use-measure";
import { useScroll, useSpring, useTransform } from "framer-motion";

export default function AboutMain({ children, ...restProps }) {
  return <AboutContainer {...restProps}>{children}</AboutContainer>;
}

AboutMain.SmallQuote = function AboutMainSmallQuote({
  children,
  ...restProps
}) {
  return <SmallQuote {...restProps}>{children}</SmallQuote>;
};

AboutMain.AboutTitle = function AboutMainAboutTitle({
  variant,
  children,
  ...restProps
}) {
  return (
    <AboutTitle variants={variant} {...restProps}>
      {children}
    </AboutTitle>
  );
};

AboutMain.MidQuote = function AboutMainMidQuote({
  variant,
  children,
  ...restProps
}) {
  return (
    <MidQuote variants={variant} {...restProps}>
      {children}
    </MidQuote>
  );
};

AboutMain.LargeQuote = function AboutMainLargeQuote({
  children,
  ...restProps
}) {
  return <LargeQuote {...restProps}>{children}</LargeQuote>;
};

AboutMain.Description = function AboutMainDescription({
  variant,
  children,
  ...restProps
}) {
  return (
    <Description variants={variant} {...restProps}>
      {children}
    </Description>
  );
};

AboutMain.TextContent = function AboutMainTextContent({
  children,
  ...restProps
}) {
  return <TextContent {...restProps}>{children}</TextContent>;
};

AboutMain.ImageFourContainer = function AboutMainImageFourContainer({
  children,
  ...restProps
}) {
  return <ImageFourContainer {...restProps}>{children}</ImageFourContainer>;
};

AboutMain.ImageFiveContainer = function AboutMainImageFiveContainer({
  children,
  ...restProps
}) {
  return <ImageFiveContainer {...restProps}>{children}</ImageFiveContainer>;
};

AboutMain.HeroStack = function AboutMainHeroStack({
  variant,
  children,
  ...restProps
}) {
  return (
    <HeroStack
      variants={variant}
      initial="hidden"
      animate="show"
      {...restProps}
    >
      {children}
    </HeroStack>
  );
};

AboutMain.ImageOne = function AboutMainImageOne({
  variant,
  children,
  ...restProps
}) {
  return (
    <ImageOne variants={variant} {...restProps}>
      {children}
    </ImageOne>
  );
};

AboutMain.ImageTwo = function AboutMainImageTwo({
  variant,
  isBigDevice,
  children,
  ...restProps
}) {
  return (
    isBigDevice && (
      <ImageTwo variants={variant} {...restProps}>
        {children}
      </ImageTwo>
    )
  );
};

AboutMain.ImageDivider = function AboutMainImageDivider({
  children,
  ...restProps
}) {
  return <ImageDivider {...restProps}>{children}</ImageDivider>;
};

AboutMain.ImageThree = function AboutMainImageThree({
  children,
  ...restProps
}) {
  return <ImageThree {...restProps}>{children}</ImageThree>;
};

AboutMain.ImageFour = function AboutMainImageFour({ children, ...restProps }) {
  return <ImageFour {...restProps}>{children}</ImageFour>;
};

AboutMain.ImageFive = function AboutMainImageFive({ children, ...restProps }) {
  return <ImageFive {...restProps}>{children}</ImageFive>;
};

AboutMain.Image = function AboutMainImage({ ...restProps }) {
  return <Image {...restProps} />;
};

AboutMain.ImageDescription = function AboutMainImageDescription({
  children,
  ...restProps
}) {
  return <ImageDescription {...restProps}>{children}</ImageDescription>;
};
