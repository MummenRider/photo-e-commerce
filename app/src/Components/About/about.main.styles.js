import { motion } from "framer-motion";
import styled from "styled-components/macro";

export const AboutContainer = styled(motion.section)`
  background: var(--bg-primary);
  padding: 2em 0em;
  color: var(--bg-secondary);
  padding-block: 160px;
  overflow-x: hidden;
  @media (min-width: 48em) {
    padding-block: 260px;
  }
`;

export const HeroStack = styled(motion.div)`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  padding: 0 1em;
  @media (min-width: 48em) {
    padding: unset;
  }
`;
export const ImageOne = styled(motion.div)`
  grid-column: 1 / span 7;
  grid-row: 1;

  @media (min-width: 48em) {
    grid-column: 2 / span 3;
  }
`;
export const ImageTwo = styled(motion.div)`
  grid-column: 8 / 12;
  grid-row: 1;
  padding-top: 25%;
  position: relative;
`;
export const AboutTitle = styled(motion.div)`
  position: absolute;
  display: block;
  font-family: var(--ff-fourth);
  font-weight: bolder;
  color: #ece4d9;
  opacity: 0.6;
  font-size: 11vw;
  letter-spacing: 0.025rem;
  top: -8%;
`;

export const ImageDivider = styled.div`
  padding: 10% 0;

  @media (min-width: 48em) {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
  }
`;

export const ImageThree = styled.div`
  grid-column: 1 / -1;
`;

export const ImageFourContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  @media (min-width: 48em) {
    grid-template-columns: repeat(24, 1fr);
  }
`;

export const ImageFour = styled.div`
  grid-column: 3 / span 8;
  @media (min-width: 48em) {
    grid-column: 6 / 14;
    grid-row: 1;
  }
`;

export const ImageFive = styled.div`
  grid-column: 3 / span 8;
  @media (min-width: 48em) {
    grid-column: 12 / 20;
    grid-row: 1;
  }
`;

export const ImageFiveContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  @media (min-width: 48em) {
    grid-template-columns: repeat(24, 1fr);
  }
`;

export const TextContent = styled.div`
  grid-column: ${(props) => props.subTitle.gridColumn};
  grid-row: ${(props) => props.subTitle.gridRow};

  h1 {
    padding-top: 1em;
    font-family: var(--ff-secondary);
    font-weight: 500;
    font-size: clamp(2.5rem, 3.125vw, 3.125rem);
    line-height: 1.2;
  }

  @media (min-width: 48em) {
    grid-column: ${(props) => props.subTitle.gridColumnB};
    grid-row: 1;
    padding-top: 20%;
    place-self: start end;
    text-align: ${(props) => props.subTitle.textAlign};
    padding-right: ${(props) => props.subTitle.paddingRight}%;
    h1 {
      padding-top: unset;
      display: inline; /* keep the background color wrapped tight */
      background-color: white;
      color: var(--bg-secondary);
      padding: 1vw;
      line-height: 1.6;
      font-size: clamp(2rem, 3.4vw, 8rem);
    }
  }
`;

export const ImageDescription = styled.p`
  grid-column: ${(props) => props.desc.gridColumn};
  grid-row: ${(props) => props.desc.gridRow};
  font-family: var(--ff-third);
  letter-spacing: 0.025rem;
  font-weight: 400;
  font-size: clamp(0.938rem, 2.5vw, 1.125rem);
  line-height: 1.5;
  padding: 1em 0;

  @media (min-width: 48em) {
    grid-column: ${(props) => props.desc.gridColumnB};
    grid-row: 2;
    font-size: clamp(0.75rem, 0.95vw, 1.8rem);

    text-align: start;
    line-height: 2;
    padding-top: 2em;
    padding-right: ${(props) => props.desc.paddingR}%;
    padding-left: ${(props) => props.desc.paddingL}%;
  }
`;

export const Image = styled.img`
  display: block;
  max-width: 100%;
  object-fit: cover;
`;
export const SmallQuote = styled.h4`
  font-family: var(--ff-third);
  letter-spacing: 0.025rem;
  font-size: clamp(0.938rem, 2.5vw, 1.125rem);
  line-height: 1.5;
  font-weight: 400;
  padding: 1em;
  max-width: 300px;
  text-align: start;
  @media (min-width: 48em) {
    font-size: clamp(0.75rem, 0.95vw, 1.8rem);
    line-height: 2;
    place-self: end;
    text-align: end;
    max-width: max(30vw, 500px);
    padding: 1em 0em 0em 0em;
    grid-column: 7 / 12;
  }
`;
export const MidQuote = styled(motion.div)`
  grid-column: 8 / -1;
  grid-row: 1;
  font-family: var(--ff-secondary);
  font-style: italic;
  font-weight: 400;
  font-size: clamp(1.125rem, 4.6vw, 2.25rem);
  line-height: 1.2;
  padding-left: 1.25em;
  position: relative;
  padding-top: 50%;
  height: fit-content;

  &::before {
    content: "\\201C";
    padding-top: 50%;
    display: block;
    font-size: clamp(2.5rem, 10.5vw, 5rem);
    position: absolute;
    font-weight: bolder;
    left: 4%;
    top: -12%;
    color: #e1c48f;
  }

  @media (min-width: 48em) {
    font-size: clamp(0.95rem, 1.8vw, 3.75rem);
    grid-column: 5 / 8;
    padding-left: 2em;
    padding-right: 1em;
    line-height: 1.45;
    &::before {
      font-size: clamp(2.5rem, 6vw, 12.5rem);
    }
  }
`;
export const LargeQuote = styled.div`
  h1 {
    font-family: var(--ff-secondary);
    text-transform: uppercase;
    font-weight: 500;
    font-style: italic;
    font-size: clamp(2.5rem, 3.125vw, 3.125rem);
    padding: 1.5em 0.5em;
  }

  @media (min-width: 48em) {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    h1 {
      grid-column: 3/11;
      font-size: clamp(3rem, 4.8vw, 10rem);
    }
  }
`;

export const Description = styled(motion.div)`
  grid-column: 4 / -1;
  grid-row: 1;
  padding-top: 90%;
  /* z-index: 300; */
  p {
    background: var(--bg-primary);
    padding: 1em;
    font-family: var(--ff-third);
    letter-spacing: 0.025rem;
    font-weight: 400;
    font-size: clamp(0.938rem, 2.5vw, 1.125rem);
    line-height: 1.5;
  }

  @media (min-width: 48em) {
    grid-column: 4 / 8;
    height: fit-content;
    padding-top: 80%;
    padding-left: 8%;

    p {
      font-size: clamp(0.75rem, 0.95vw, 1.8rem);
      padding: 3.125em;
      line-height: 2;
    }
  }
`;
