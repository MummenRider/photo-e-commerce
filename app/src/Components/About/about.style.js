import styled from "styled-components";

export const DescriptionTitle = styled.h1`
  font-size: 4rem;
  text-align: center;
  font-weight: 300;
  padding: 4rem 0;
  @media screen and (min-width: 768px) {
    padding: 0;
  }

  @media screen and (min-width: 1224px) {
    font-size: 5rem;
    font-weight: 300;
  }

  @media screen and (min-width: 1224px) {
    font-size: 7rem;
    font-weight: 300;
  }
`;

export const DescriptionParagraph = styled.p`
  font-size: 1.4rem;
  font-weight: 200;
  padding: 1rem 0;
  line-height: 20px;
  text-align: justify;
  @media screen and (min-width: 1224px) {
    font-size: 1.6rem;
    font-weight: 100;
    line-height: 26px;
  }

  @media screen and (min-width: 1440px) {
    font-size: 1.8rem;
    line-height: 30px;
  }
`;

export const AboutImage = styled.div`
  height: 80vh;
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
`;

export const ImageContent = styled.div`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow: hidden;
  background-image: url(${(props) => props.bkgImg});
  background-size: cover;
  background-repeat: no-repeat;

  @media screen and (min-width: 768px) {
    filter: brightness(75%);
    background-position: 25% 100%;
  }
`;

export const AboutWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: auto;
  @media screen and (min-width: 768px) {
    flex-direction: row;
    height: 100vh;
    min-height: 912px;
    padding: 8rem 0;
    width: 80%;
    margin: auto;
    gap: 30px;
    > * {
      flex-basis: 100%;
    }
    & ${DescriptionTitle}:nth-child(1) {
      display: none;
    }
  }

  @media screen and (min-width: 1268px) {
    gap: 100px;
  }
`;

export const AboutDescription = styled.div`
  width: 80%;
  margin: 0 auto;
  & ${DescriptionTitle}:nth-child(1) {
    display: none;
  }

  & ${DescriptionParagraph}:nth-child(2) {
    padding-top: 3rem;
  }

  @media screen and (min-width: 768px) {
    & ${DescriptionTitle}:nth-child(1) {
      display: block;
    }
  }
`;

export const AboutContainer = styled.section`
  background-color: #111112;
  font-family: "Montserrat", sans-serif;
  color: white;

  @media screen and (min-width: 768px) {
    height: 100vh;
    min-height: 912px;
  }
`;

export const AboutCircleImage = styled.div``;
