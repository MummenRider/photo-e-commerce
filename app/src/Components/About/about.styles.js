import styled from "styled-components";
import { motion } from "framer-motion";

export const Container = styled.section`
  width: 100%;
  font-family: "Montserrat", sans-serif;
  color: white;
  /* display: flex;
  flex-direction: column;

  @media screen and (min-width: 768px) {
    flex-direction: row;
  } */
`;

export const Left = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Right = styled.div`
  margin: 0 auto;
  padding: 5em 10%;

  @media screen and (min-width: 1000px) {
    padding: 20% 0%;
  }
`;

export const Inner = styled.div`
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 1300px) {
    flex-direction: row;
    padding: 8rem 0;
    width: 80%;
    margin: 0 auto;
    gap: 10%;
  }
`;

export const Frames = styled(motion.div)`
  width: 100vw;
  @media screen and (min-width: 1300px) {
    width: 30vw;
  }
`;

export const Image = styled(motion.div)`
  background-image: url(${(props) => props.url});
  width: 100%;
  background-size: cover;
  background-repeat: no-repeat;
  padding-top: 98.5981%;
`;

export const Content = styled.div`
  width: 80%;
  margin: auto;

  @media screen and (max-width: 768px) {
    padding: 6rem 0;
  }
`;
export const Title = styled.h3`
  font-size: clamp(4rem, 5vw, 36rem);
  text-align: center;
  font-weight: 300;
  padding: 10rem 0;
  @media screen and (min-width: 1300px) {
    text-align: left;
    padding: 2rem 0;
  }
`;

export const Description = styled(motion.p)`
  font-size: clamp(1.5rem, 0.9vw, 6rem);
  font-weight: 100;
  padding: 1rem 0;
  line-height: 1.8;
  text-align: left;
`;
