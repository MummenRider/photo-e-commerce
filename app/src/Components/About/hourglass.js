export function Hourglass() {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="6vw"
      height="6vh"
      viewBox="0 0 36 83.329"
    >
      <g transform="translate(-893.062 -3019.773)">
        <path
          class="a"
          d="M12.636,0c8.645,0,15.653,8.059,15.653,18S21.281,36,12.636,36-13.875,18-13.875,18,3.992,0,12.636,0Z"
          transform="translate(893.062 3048.062) rotate(-90)"
        />
        <path
          d="M12.636,0c8.645,0,15.653,8.059,15.653,18S21.281,36,12.636,36-13.875,18-13.875,18,3.992,0,12.636,0Z"
          transform="translate(929.062 3074.813) rotate(90)"
        />
      </g>
    </svg>
  );
}
