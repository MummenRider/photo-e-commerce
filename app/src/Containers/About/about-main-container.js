import AboutMain from "Components/About/about.main";
import One from "Assets/Images/nico-2.png";
import Two from "Assets/Images/abt-nico.png";
import Three from "Assets/Images/about-1.jpg";
import Four from "Assets/Images/about-2.png";
import Five from "Assets/Images/about-3.png";
import { useMediaQuery } from "react-responsive";
import { NavbarContainer } from "../../Containers/Home/navbar-container";
import { motion, useInView } from "framer-motion";
import { useEffect, useRef } from "react";
import TextAnimation from "Components/Animation/text-animation";
export function AboutMainContainer({isFirstMount}) {
  const isBigDevice = useMediaQuery({ query: "(min-width: 48em)" });
  const fourSubTitle = {
    gridColumn: "3 / span 8",
    gridRow: 2,
    gridColumnB: "12 / 20",
    paddingRight: 0,
    textAlign: "end",
  };
  const fourDescription = {
    gridColumn: "3 / span 9",
    gridRow: 3,
    gridColumnB: "6 / 12",
    paddingR: 0,
    paddingL: 40,
  };

  const fiveSubTitle = {
    gridColumn: "3 / span 8",
    gridRow: 2,
    gridColumnB: "6 / span 7",
    textAlign: "start",
  };
  const fiveDescription = {
    gridColumn: "3 / span 9",
    gridRow: 3,
    gridColumnB: "6 / span 7",
    paddingR: 30,
    paddingL: 0,
  };
  const parentVariant = {
    hidden: {
      opacity: 0,
    },
    show: {
      opacity: 1,
      transition: {
        when: "beforeChildren",
        staggerChildren: 0.3,
        ease: "easeInOut",
      },
    },
  };

  const childVariant = {
    hidden: { opacity: 1, y: 30 },
    show: {
      opacity: 1,
      y: 0,

      transition: {
        duration: 0.35,
        ease: [0.12, 0, 0.39, 0],
      },
    },
  };

  const childVariant1 = {
    hidden: { opacity: 0, y: -8 },
    show: {
      y: 0,
      opacity: 1,
      transition: {
        duration: 1,
        delay: 0.9,
        ease: [0.12, 0, 0.39, 0],
      },
    },
  };
  // const textRef = useRef(null);
  // const isInView = useInView(textRef, {
  //   once: false,
  //   amount: "some",
  // });


  return (
    <AboutMain>
      <AboutMain.HeroStack isFirstMount={isFirstMount} variant={parentVariant}>
        <NavbarContainer
          variants={childVariant1}
          color="#313B41"
          bgColor="var(--bg-primary)"
        />
        <AboutMain.ImageOne variant={childVariant}>
          <AboutMain.Image src={One} />
        </AboutMain.ImageOne>
        <AboutMain.ImageTwo isBigDevice={isBigDevice} variant={childVariant}>
          <AboutMain.Image src={Two} />
          <AboutMain.AboutTitle variant={childVariant1}>
            ABOUT
          </AboutMain.AboutTitle>
        </AboutMain.ImageTwo>
        <AboutMain.MidQuote variant={childVariant1}>
          Some photography quotes here this is another line, extra, and one more
          here
        </AboutMain.MidQuote>
        <AboutMain.Description variant={childVariant1}>
          <p>
            Iâ€™m a photographer based in Regina, Saskatchewan. I started
            photography when I was in high school, and I totally fell in love
            with it. Until now, I can still remember the day that my photography
            teacher handed me her camera.
          </p>
        </AboutMain.Description>
      </AboutMain.HeroStack>
      <AboutMain.ImageDivider>
        <AboutMain.ImageThree>
          <AboutMain.Image src={Three} />
        </AboutMain.ImageThree>

        <AboutMain.SmallQuote>
          <TextAnimation once={true}>
            How you got here in one/two sentence(s) -Lorem ipsum dolor sit amet,
            consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt
            ut labore et dolore magna aliquyam erat, sed diam voluptua.
          </TextAnimation>
        </AboutMain.SmallQuote>
      </AboutMain.ImageDivider>
      <AboutMain.ImageFourContainer>
        <AboutMain.ImageFour>
          <AboutMain.Image src={Four} />
        </AboutMain.ImageFour>
        <AboutMain.TextContent
          subTitle={fourSubTitle}
          isBigDevice={isBigDevice}
        >
          <TextAnimation>
            <h1>What inspires me to be creative in my work</h1>
          </TextAnimation>
          <TextAnimation animateDirection="y" amount={0.2}>
            <AboutMain.ImageDescription desc={fourDescription}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris
              nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
              reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur.
            </AboutMain.ImageDescription>
          </TextAnimation>
        </AboutMain.TextContent>
      </AboutMain.ImageFourContainer>
      <AboutMain.LargeQuote>
        <h1>In one sentence, what makes you unique to other photographers</h1>
      </AboutMain.LargeQuote>
      <AboutMain.ImageFiveContainer>
        <AboutMain.ImageFive>
          <AboutMain.Image src={Five} />
        </AboutMain.ImageFive>
        <AboutMain.TextContent subTitle={fiveSubTitle}>
          <TextAnimation reverse={true}>
            <h1>Where I'm looking to go next</h1>
          </TextAnimation>
          <TextAnimation animateDirection="y" amount={0.2}>
            <AboutMain.ImageDescription desc={fiveDescription}>
              Sed ut perspiciatis unde omnis iste natus error sit voluptatem
              accusantium doloremque laudantium, totam rem aperiam, eaque ipsa
              quae ab illo inventore veritatis et quasi architecto beatae vitae
              dicta sunt explicabo.
            </AboutMain.ImageDescription>
          </TextAnimation>
        </AboutMain.TextContent>
      </AboutMain.ImageFiveContainer>
    </AboutMain>
  );
}
