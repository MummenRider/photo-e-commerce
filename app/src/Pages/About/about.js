import { AboutMainContainer } from "Containers/About/about-main-container";
import { motion } from "framer-motion";

import { FooterContainer } from "Containers/Home/footer-container";
import { PageTransition } from "../../Components/Transitions/transition";
import { useEffect, useRef } from "react";
export default function About({isFirstMount}) {
console.log('About', isFirstMount);
  return (
    <>
      <PageTransition>
        <AboutMainContainer isFirstMount={isFirstMount} />
        <FooterContainer />
      </PageTransition>
    </>
  );
}
