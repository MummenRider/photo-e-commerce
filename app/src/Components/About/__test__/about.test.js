import React from "react";
import { cleanup, getByText, render, screen } from "@testing-library/react";
import About from "Components/about/about";
import { mockAllIsIntersecting } from "react-intersection-observer/test-utils";
afterEach(cleanup);
describe("<About />", () => {
  it("should render <About />", () => {
    const { container, getByText, queryByText } = render(
      <About>
        <About.Title isBigDevice={true}>Title here</About.Title>
        <About.Description>Some description</About.Description>
      </About>
    );
    expect(queryByText("Title here")).toBeTruthy();
    expect(getByText("Some description")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render <About /> with title hidden to user", () => {
    const { container, queryByText } = render(
      <About>
        <About.Title isBigDevice={false}>Title here</About.Title>
      </About>
    );
    expect(queryByText("Title here")).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
