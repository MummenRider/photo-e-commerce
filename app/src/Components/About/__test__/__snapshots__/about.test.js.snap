// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render <About /> 1`] = `
<section
  class="sc-bdnxRM dazqpe"
>
  <div
    class="sc-hKFxyN jelMKq"
  >
    <h3
      class="sc-iCoGMd lebLBC"
    >
      Title here
    </h3>
    <p
      class="sc-fujyAs iFKMfc"
    >
      Some description
    </p>
  </div>
</section>
`;

exports[`<About /> should render <About /> with title hidden to user 1`] = `
<section
  class="sc-bdnxRM dazqpe"
>
  <div
    class="sc-hKFxyN jelMKq"
  />
</section>
`;
